src/storage/kv.rs

```diff
@@ -279,6 +324,9 @@ impl LogKeyValueStore {
 
         self.current_height.increment()?;
 
+        self.key_pointer_map = HashMap::new();
```

Comment:
I think we should cancel `key_pointer_map`. If we call them `snapshot` we always use the `snapshot` name.

Reply:
good point, I think a better way to solve this is extract `SnapShot ` as a real struct, it should have `cancel` method in it.
I will do it in another PR.